### PR TITLE
Read many with user passed session token coverage and logging WARN for incorrectly mapped session token to pkRangeId

### PR DESCRIPTION
**Pull request by jeet1995**
This PR introduces the following changes:
1. Introduce code coverage for the `readMany` overload which accepts the `sessionToken`.
2. Add a `warn`-level log and a note in `CosmosDiagnostics` to print / note cases where the `pkRangeId` segment of the user-passed `sessionToken` doesn't map to the `resolvedPartitionKeyRangeId` for the request. This is done so because the SDK silently drops the session token otherwise - this makes the request use `Eventual` consistency and can result in 404 / 0 errors. Adding a `warn`-level log can help analyze application logs when such an issue occurs.

**NOTE :** In Gateway connectivity mode, the SDK doesn't drop the session token (behavior matched with that of the .NET SDK) - this issue would get pushed down to the Gateway's SDK client which is a Direct connectivity mode SDK - worth following up here. In other words, this logging only applies to the Java SDK in Direct connectivity mode.

Diagnostic change:

image

**WARN** log is enabled by default, can be disabled through the following system property:

```java
System.setProperty("COSMOS.SHOULD_LOG_INCORRECTLY_MAPPED_USER_SESSION_TOKEN", "false");
```

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 1]
```diff
@@ -5,16 +5,17 @@
...
 #### Breaking Changes
-* Fixed an issue where customer provided session token is not honored for the `readMany` operation. - See[PR 39265](https://github.com/Azure/azure-sdk-for-java/pull/39265)
 
 #### Bugs Fixed
 * Fixed an issue where `contactedRegions` shows the wrong region in a multi region account if no preferred regions are specified. - See [PR 41045](https://github.com/Azure/azure-sdk-for-java/pull/41045)
+* Fixed an issue where customer provided session token is not honored for the `readMany` operation. - See [PR 39265](https://github.com/Azure/azure-sdk-for-java/pull/39265)
 
 #### Other Changes
 * Added metrics and tracing for ReadMany operations. - See [PR 41042](https://github.com/Azure/azure-sdk-for-java/pull/41042)
+* Added a `warn`-level log to capture when the `pkRangeId` in the user-passed session token and the `resolvedPartitionKeyRangeId` in the request doesn't match. -  See [PR 41268](https://github.com/Azure/azure-sdk-for-java/pull/41268)
```
Comment (tvaron3) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:18]:
```suggestion
* Added a `warn`-level log to capture when the `pkRangeId` in the user-passed session token and the `resolvedPartitionKeyRangeId` in the request doesn't match. - See [PR 41268](https://github.com/Azure/azure-sdk-for-java/pull/41268)
```

**Review by tvaron3 — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos/CHANGELOG.md (hunk 1 above)
Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: Fixed in next iteration.

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenHelper.java [hunk 2]
```diff
@@ -108,7 +112,16 @@ private static ISessionToken getLocalSessionToken(
                 } else {
                     highestSessionToken = highestSessionToken.merge(parsedSessionToken);
                 }
+            }
+        }
 
+        if (highestSessionToken == null) {
+            if (StringUtils.isNotEmpty(globalSessionToken)) {
+                logger.warn("The session token : {} for partition key range id : {} and collection rid : {} could not be evaluated, " +
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenHelper.java:120]: I am thinking usually this should only happen when it is the first time SDK sending requests to that partitionKeyRangeId? so not sure warning level is the one to be used here. 

And also in the diagnostics, we also log the request sessionToken SDK sent to the backend, will it be enough for debugging purpose?

Comment (jeet1995) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenHelper.java]: Discussed offline.

**Pull request merged**